app/test_main.py

```diff
@@ -1,3 +1,29 @@
+import pytest
+
 from app.main import get_human_age
 
-# write your code here
+
+@pytest.mark.parametrize(
+    "cat_age,dog_age,result",
+    [
+        pytest.param(0, 0, [0, 0], id="when input is zero, output "
+                                      "should be too"),
+        pytest.param(12, 12, [0, 0], id="when animal`s years are less "
+                                        "than 15, output should be "
+                                        "equal to zero"),
+        pytest.param(15, 15, [1, 1], id="output of this limit values "
+                                        "should be equal to 1"),
+        pytest.param(23, 23, [1, 1], id="output should be equal to 1 year"),
+        pytest.param(24, 24, [2, 2], id="output of this limit values "
+                                        "should be equal to 2"),
+        pytest.param(27, 27, [2, 2], id="output of this limit value "
+                                        "for cat equals 2 and for dog too"),
+        pytest.param(28, 28, [3, 2], id="output of this limit value "
+                                        "for cat equals 3 but for dog 2"),
+        pytest.param(29, 29, [3, 3], id="output of this limit value "
+                                        "for cat equals 3 and for dog too"),
+        pytest.param(100, 100, [21, 17], id="checking big values")
+    ]
+)
```

Comment:
missing some checklist requirements: handling negative numbers and raising errors

Reply:
But do I have the right to change the two main functions from the main.py?  Because no exceptions are raised there, and accordingly the test will always fail.

Reply:
You don’t have to. Just make test where you will pass incorrect parameters and test for exception being raised

Reply:
Due to the fact that the main function does not pass this test, Github shows me that all checks have failed

Reply:
```
@pytest.mark.parametrize(
    "cat_age,dog_age,expected_error",
    [
        pytest.param(-2,
                     -36,
                     ValueError,
                     id="function should raise error if input is negative"),

        pytest.param(987654321,
                     1000000,
                     ValueError,
                     id="function should raise "
                        "error if input is large numbers")
    ]
)
def test_get_human_age_errors(cat_age: int,
                              dog_age: int,
                              expected_error: Exception) -> None:
    with pytest.raises(expected_error):
        get_human_age(cat_age, dog_age)

```

Reply:
There will be no valueerror here because you are passing two ints. the function may be cause TypeError if one of the arguments will be not a numeric type

Reply:
But what type of exception should I use here? If ValueError is incorrect and TypeError for values which are not numeric and I have only numeric input...

Reply:
You can pass strings as params and expect TypeError out of it for example

Reply:
I'm so sorry. But how I should check negative numbers?

Reply:
you've done it correct - neg numbers will return 0,0 and no error